src/api/v1/user.go

```diff
@@ -0,0 +1,119 @@
+package v1
+
+import (
+	"github.com/ImpactDevelopment/ImpactServer/src/database"
+	"github.com/ImpactDevelopment/ImpactServer/src/middleware"
+	"github.com/labstack/echo/v4"
+	"log"
+	"net/http"
+)
+
+func getUser(c echo.Context) error {
+	if user := middleware.GetUser(c); user != nil {
+		return c.JSON(http.StatusOK, user)
+	} else {
+		return echo.NewHTTPError(http.StatusInternalServerError, "unable to cast user")
+	}
+}
+
+func patchUser(c echo.Context) error {
+	if user := middleware.GetUser(c); user != nil {
+		// Everything is a pointer so we can check what was present in the request
+		// e.g. an unset field defaulting to false might be bad
+		var body struct {
+			Email         *string `json:"email"`
+			Minecraft     *string `json:"minecraft"`
+			DiscordToken  *string `json:"discord"`
+			Password      *string `json:"password"`
+			LegacyEnabled *bool   `json:"legacy_enabled"`
+			Incognito     *bool   `json:"incognito"`
```

Comment:
I changed it to incognito because the behaviour changed from hiding capes to hiding the user from user_info. Iirc other stuff is also hidden like nametag infoband icons. The _goal_ of the option is not to hide capes, the goal is to be _incognito_